File: src/com/iotracks/iofabric/utils/BytesUtil.java
package com.iotracks.iofabric.utils;

import java.nio.ByteBuffer;

public class BytesUtil {
	
    public static byte[] longToBytes(long x) {
    	ByteBuffer buffer;    
    	buffer = ByteBuffer.allocate(Long.BYTES);
    	buffer.putLong(0, x);
    	byte[] result = buffer.array();
    	buffer.clear();
        return result;
    }

    public static long bytesToLong(byte[] bytes) {
    	if (bytes.length < Long.BYTES)
    		return 0;
    	ByteBuffer buffer;    
    	buffer = ByteBuffer.allocate(Long.BYTES);
        buffer.put(bytes, 0, bytes.length);
        buffer.flip();//need flip
        long result = buffer.getLong();
        buffer.clear();
        return result;
    }
    
    public static byte[] integerToBytes(int x) {
    	ByteBuffer buffer;    
    	buffer = ByteBuffer.allocate(Integer.BYTES);
        buffer.putInt(0, x);
    	byte[] result = buffer.array();
    	buffer.clear();
        return result;
    }

    public static int bytesToInteger(byte[] bytes) {
    	if (bytes.length < Integer.BYTES)
    		return 0;
    	ByteBuffer buffer;    
    	buffer = ByteBuffer.allocate(Integer.BYTES);
        buffer.put(bytes, 0, bytes.length);
        buffer.flip();//need flip
        int result = buffer.getInt();
        buffer.clear();
        return result;
    }
    
    public static byte[] shortToBytes(short x) {
    	ByteBuffer buffer;    
    	buffer = ByteBuffer.allocate(Short.BYTES);
        buffer.putShort(0, x);
    	byte[] result = buffer.array();
    	buffer.clear();
        return result;
    }

    public static short bytesToShort(byte[] bytes) {
    	if (bytes.length < Short.BYTES)
    		return 0;
    	ByteBuffer buffer;    
    	buffer = ByteBuffer.allocate(Short.BYTES);
        buffer.put(bytes, 0, bytes.length);
        buffer.flip();//need flip
        short result = buffer.getShort();
        buffer.clear();
        return result;
    }
    
    public static byte[] doubleToBytes(int x) {
    	ByteBuffer buffer;    
    	buffer = ByteBuffer.allocate(Double.BYTES);
        buffer.putDouble(0, x);
    	byte[] result = buffer.array();
    	buffer.clear();
        return result;
    }

    public static double bytesToDouble(byte[] bytes) {
    	if (bytes.length < Double.BYTES)
    		return 0;
    	ByteBuffer buffer;    
    	buffer = ByteBuffer.allocate(Double.BYTES);
        buffer.put(bytes, 0, bytes.length);
        buffer.flip();//need flip
        double result = buffer.getDouble();
        buffer.clear();
        return result;
    }
    
    public static byte[] stringToBytes(String s) {
    	if (s == null)
    		return new byte[] {};
    	else
    		return s.getBytes();
    }
    
    public static String bytesToString(byte[] bytes) {
        return new String(bytes);
    }

    public static String byteArrayToString(byte[] bytes) {
    	StringBuilder result = new StringBuilder();
    	
		result.append("[");
    	for (byte b : bytes) {
    		if (result.length() > 1)
    			result.append(", ");
    		result.append(b);
    	}
		result.append("]");
    	
    	return result.toString();
    }
}

File: src/ContainerSdk/Utils/BytesUtil.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Utils
{
	public class BytesUtil
	{
		public static byte[] CopyRange(byte[] src, int from, int to)
		{
			byte[] temp = new byte[from];
			byte[] result = new byte[to - from];
			MemoryStream input = new MemoryStream();
			input.Read(temp, 0, temp.Length);
			input.Read(result, 0, result.Length);
			try
			{
				input.Close();
			}
			catch
			{

			}

			return result;
		}

		public static byte[] LongToBytes(long x)
		{
			byte[] b = new byte[8];
			for (int i = 0; i < 8; ++i)
			{
				b[i] = (byte)(x >> (8 - i - 1 << 3));
			}

			return b;
		}

		public static long BytesToLong(byte[] bytes)
		{
			long result = 0;
			for (int i = 0; i < bytes.Length; i++)
			{
				result = (result << 8) + (bytes[i] & 0xff);
			}

			return result;
		}

		public static byte[] IntegerToBytes(int x)
		{
			byte[] b = new byte[4];
			for (int i = 0; i < 4; ++i)
			{
				b[i] = (byte)(x >> (4 - i - 1 << 3));
			}

			return b;
		}

		public static int BytesToInteger(byte[] bytes)
		{
			int result = 0;
			for (int i = 0; i < bytes.Length; i++)
			{
				result = (result << 8) + (bytes[i] & 0xff);
			}

			return result;
		}

		public static byte[] ShortToBytes(short x)
		{
			byte[] b = new byte[2];
			for (int i = 0; i < 2; ++i)
			{
				b[i] = (byte)(x >> (2 - i - 1 << 3));
			}

			return b;
		}

		public static short BytesToShort(byte[] bytes)
		{
			short result = 0;
			for (int i = 0; i < bytes.Length; i++)
			{
				result = (short) ((result << 8) + (bytes[i] & 0xff));
			}

			return result;
		}

		public static byte[] StringToBytes(string x)
		{
			if (x == null)
			{
				return new byte[] { };
			}

			else
			{
				return Encoding.ASCII.GetBytes(x);
			}
		}

		public static string BytesToString(byte[] bytes)
		{
			string s = Encoding.UTF8.GetString(bytes);
			return s;
		}

		public static string ByteArrayToString(byte[] bytes)
		{
			StringBuilder result = new StringBuilder();
			result.Append("[");
			for (int i = 0; i < bytes.Length; i++)
			{
				if (bytes.Length > 1)
				{
					result.Append(", ");
				}

				result.Append(bytes[i]);
			}
			result.Append("]");

			return result.ToString();
		}

		public static int getLength(String s)
		{
			if (s == null)
			{
				return 0;
			}

			else
			{
				return s.Length;
			}
		}
	}
}
